packages/gatsby/src/commands/develop.ts

```diff
@@ -1,631 +1,240 @@
-import queryWatcher from "../query/query-watcher"
-import requiresWriter from "../bootstrap/requires-writer"
-import {
-  reportWebpackWarnings,
-  structureWebpackErrors,
-} from "../utils/webpack-error-utils"
-import { waitUntilAllJobsComplete } from "../utils/wait-until-jobs-complete"
-import {
-  userPassesFeedbackRequestHeuristic,
-  showFeedbackRequest,
-} from "../utils/feedback"
-
-import { IPreparedUrls, prepareUrls } from "../utils/prepare-urls"
-import { Stage, IProgram } from "./types"
-
-// checks if a string is a valid ip
-const REGEX_IP = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$/
-
-// const isInteractive = process.stdout.isTTY
-
-// Watch the static directory and copy files to public as they're added or
-// changed. Wait 10 seconds so copying doesn't interfere with the regular
-// bootstrap.
-setTimeout(() => {
-  syncStaticDir()
-}, 10000)
-
-onExit(() => {
-  telemetry.trackCli(`DEVELOP_STOP`)
-})
-
-type ActivityTracker = any // TODO: Replace this with proper type once reporter is typed
-
-interface IServer {
-  compiler: webpack.Compiler
-  listener: http.Server | https.Server
-  webpackActivity: ActivityTracker
-}
+import tmp from "tmp"
+import { spawn } from "child_process"
+import chokidar from "chokidar"
+import getRandomPort from "get-port"
+import socket from "socket.io"
+import fs from "fs-extra"
+import { isCI } from "gatsby-core-utils"
+import { createServiceLock } from "gatsby-core-utils/dist/service-lock"
+import { startDevelopProxy } from "../utils/develop-proxy"
+import { IProgram } from "./types"
+
+// Adapted from https://stackoverflow.com/a/16060619
+const requireUncached = (file: string): any => {
+  try {
+    delete require.cache[require.resolve(file)]
+  } catch (e) {
+    return null
+  }
 
-async function startServer(program: IDevelopArgs): Promise<IServer> {
-  const indexHTMLActivity = report.phantomActivity(`building index.html`, {})
-  indexHTMLActivity.start()
-  const directory = program.directory
-  const directoryPath = withBasePath(directory)
-  const workerPool = WorkerPool.create()
-  const createIndexHtml = async (activity: ActivityTracker): Promise<void> => {
-    try {
-      await buildHTML({
-        program,
-        stage: Stage.DevelopHTML,
-        pagePaths: [`/`],
-        workerPool,
-        activity,
-      })
-    } catch (err) {
-      if (err.name !== `WebpackError`) {
-        report.panic(err)
-        return
-      }
-      report.panic(
-        report.stripIndent`
-          There was an error compiling the html.js component for the development server.
+  try {
+    return require(file)
+  } catch (e) {
+    return null
+  }
+}
 
-          See our docs page on debugging HTML builds for help https://gatsby.dev/debug-html
-        `,
-        err
-      )
+// Heuristics for gatsby-config.js, as not all changes to it require a full restart to take effect
+const doesConfigChangeRequireRestart = (
```

Comment:
Regexes aren't being converted to json either so this might be buggy. Nothing breaking

```js
JSON.stringify({ reg: /test$/ })
```
=> `"{"reg":{}}"`